internal/test/test.go

```diff
@@ -55,43 +62,80 @@ func run(
 	_ command.GlobalFlags,
 	services *services.Services,
 ) (command.Result, error) {
+	testFiles := make(map[string][]byte, 0)
+	for _, filename := range args {
+		code, err := readerWriter.ReadFile(filename)
 
-	filename := args[0]
+		if err != nil {
+			return nil, fmt.Errorf("error loading script file: %w", err)
+		}
 
-	code, err := readerWriter.ReadFile(filename)
-	if err != nil {
-		return nil, fmt.Errorf("error loading script file: %w", err)
+		testFiles[filename] = code
+	}
+
+	var coverageReport *runtime.CoverageReport
+	if testFlags.Cover {
+		coverageReport = runtime.NewCoverageReport()
 	}
 
 	result, err := services.Tests.Execute(
-		code,
-		filename,
+		testFiles,
 		readerWriter,
+		coverageReport,
 	)
 
 	if err != nil {
 		return nil, err
 	}
 
+	if coverageReport != nil {
+		file, err := json.MarshalIndent(coverageReport, "", "  ")
+		if err != nil {
+			return nil, fmt.Errorf("error serializing coverage report: %w", err)
+		}
+
+		err = os.WriteFile(testFlags.CoverProfile, file, 0644)
+		if err != nil {
+			return nil, fmt.Errorf("error writing coverage report file: %w", err)
+		}
+	} else if Cmd.Flags().Changed("coverprofile") {
+		fmt.Printf("%s The '--coverprofile' flag requires the '--cover' flag\n", output.WarningEmoji())
```

Comment:
return an error here

Reply:
Nice! Fixed in https://github.com/onflow/flow-cli/pull/827/commits/cc19c3c42b4620a0a637c2197c093942934bb488